src/client/gui/lib/update_available.dart

```diff
@@ -15,14 +17,35 @@ class UpdateNotifier extends Notifier<UpdateInfo> {
 
   void set(UpdateInfo updateInfo) {
     if (updateInfo.version.isBlank) return;
+
     final updateNotificationExists = ref.read(notificationsProvider).any((n) {
       return n is UpdateAvailableNotification && n.updateInfo == updateInfo;
     });
     if (updateNotificationExists) return;
+
+    // In-app notification
     ref
         .read(notificationsProvider.notifier)
         .add(UpdateAvailableNotification(updateInfo));
+
+    // Update the state
     state = updateInfo;
+
+    // Create and show a local notification
+    _showLocalNotification(updateInfo);
+  }
+
+  void _showLocalNotification(UpdateInfo updateInfo) {
+    if (!mpPlatform.showLocalUpdateNotifications) return;
+
+    final notification = LocalNotification(
+      title: 'Multipass Update Available',
+      body: 'Version ${updateInfo.version} is available. Click to upgrade now.',
+    );
+
+    notification.onClick = launchInstallUrl;
```

Comment:
Oof, I thought that the notification will automatically close when you click it, but it doesn't, so it must be closed manually.
```suggestion
    notification.onClick = () async {
      await launchInstallUrl();
      await notification.close();
    };
```